src/libraries/System.Console/src/System/TermInfo.cs

```diff
@@ -498,9 +498,8 @@ private static string ReadString(byte[] buffer, int pos)
             /// <summary>Finds the null-terminator for a string that begins at the specified position.</summary>
             private static int FindNullTerminator(byte[] buffer, int pos)
             {
-                int termPos = pos;
-                while (termPos < buffer.Length && buffer[termPos] != '\0') termPos++;
-                return termPos;
```

Comment:
Just wondering, were these detected using any kind of analysis?

Is there potential for an analyzer here?

Reply:
These were just searching around for various patterns. I could have sworn we had an issue for considering an analyzer, though I can't find it now.